lib/install.js

```diff
@@ -208,7 +208,7 @@ function Installer (where, dryrun, args) {
   this.todo = []
   this.progress = {}
   this.noPackageJsonOk = !!args.length
-  this.topLevelLifecycles = !args.length
+  this.topLevelLifecycles = !!args.length
```

Comment:
Because preinstall was never being called at all without this and I think it was a bug to begin with given how other args.length are being handled. This is the code in reference that was always triggered with args.length being the way it was https://github.com/palmerj3/npm/blob/0ee8d4248b4294c347923b8cff41f89dfe597ac9/lib/install.js#L541